src/main/java/FrameworksDrivers/ChatView.java

```diff
@@ -151,29 +142,23 @@ public void updatePage(Object[] info) {
                             currChatPanel, textMessage);
                     currChatPanel.revalidate();
                     currChatPanel.repaint();
-                    this.panel.remove(this.chatContainer);
-                    this.chatContainer = new JPanel();
-                    this.chatContainer.setLayout(this.chatLayout);
-                    this.chatContainer.setBackground(Color.YELLOW);
-                    this.chatContainer.setBounds(0, 0, 400, 600);
-
-                    this.panel.add(chatContainer, "bob joe");
-                    this.chatContainer.add(currChatPanel, String.valueOf(index));
-                    this.map.put(currChatPanel, String.valueOf(index));
-                    this.chatContainer.revalidate();
-                    this.chatContainer.repaint();
                 }
+                this.panel.remove(this.chatContainer);
```

Comment:
I was dumb and had put the painting of chats inside the for loop,

Reply:
Can you elaborate on this? I don't really understand why you need to paint/repaint inside or outside the for loop. We can discuss this on discord or in-person.